pkg/webui/console/containers/device-onboarding-form/device-onboarding-form.js

```diff
@@ -1,67 +0,0 @@
-// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-import React, { useCallback } from 'react'
-import { merge } from 'lodash'
-
-import Form from '@ttn-lw/components/form'
-import SubmitBar from '@ttn-lw/components/submit-bar'
-import SubmitButton from '@ttn-lw/components/submit-button'
-
-import sharedMessages from '@ttn-lw/lib/shared-messages'
-import PropTypes from '@ttn-lw/lib/prop-types'
-
-import DeviceProvisioningFormSection, {
-  initialValues as provisioningInitialValues,
-} from './device-provisioning-form-section'
-import DeviceTypeFormSection, {
-  initialValues as typeInitialValues,
-} from './device-type-form-section'
-import validationSchema from './validation-schema'
-
-const initialValues = merge({}, provisioningInitialValues, typeInitialValues)
-
-const DeviceOnboardingForm = props => {
-  const { appId, getRegistrationTemplate } = props
-  const handleSubmit = useCallback(async values => {
-    console.log(values)
-  }, [])
-
-  return (
-    <Form
-      onSubmit={handleSubmit}
-      initialValues={initialValues}
-      validationSchema={validationSchema}
-      stripUnusedFields
-    >
-      <DeviceTypeFormSection appId={appId} getRegistrationTemplate={getRegistrationTemplate} />
```

Comment:
Instead of passing state-related props through as component props, it is preferable to do the necessary selection in the child component itself. Keep in mind that being able to avoid prop waterfalls is one of the main reasons to use a global state engine like redux.